haystack_experimental/components/embedders/image/sentence_transformers_doc_image_embedder.py

```diff
@@ -0,0 +1,361 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import mimetypes
+from collections import defaultdict
+from copy import copy
+from pathlib import Path
+from typing import Any, Dict, List, Literal, Optional, Tuple, TypedDict, Union
+
+from haystack import Document, component, default_from_dict, default_to_dict
+from haystack.components.embedders.backends.sentence_transformers_backend import (
+    _SentenceTransformersEmbeddingBackend,
+    _SentenceTransformersEmbeddingBackendFactory,
+)
+from haystack.dataclasses.byte_stream import ByteStream
+from haystack.lazy_imports import LazyImport
+from haystack.utils.auth import Secret, deserialize_secrets_inplace
+from haystack.utils.device import ComponentDevice
+from haystack.utils.hf import deserialize_hf_model_kwargs, serialize_hf_model_kwargs
+from typing_extensions import NotRequired
+
+from haystack_experimental.components.image_converters.image_utils import (
+    _convert_pdf_to_pil_images,
+    _resize_image_preserving_aspect_ratio,
+)
+from haystack_experimental.dataclasses.image_content import IMAGE_MIME_TYPES
+
+with LazyImport("Run 'pip install pillow'") as pillow_import:
+    from PIL import Image as PILImage
+    from PIL.Image import Image
+    from PIL.ImageFile import ImageFile
+
+
+class ImageSourceInfo(TypedDict):
```

Comment:
Is there a way to make this private?

Reply:
I thought about this point.
If we don't export this to `__init__.py` and we only use it in internal methods I would not expect users to try to use it.
But if we want to be more explicit, we can rename it to `_ImageSourceInfo`

Reply:
Hmm I think we if we'd really like it to be private we should make it that way. Another way I've seen libraries do this (e.g. datadog) is to make the import path include a `_`. So we could do something like `../image/_types.py` or some other file name to indicate things inside that file are private.

Reply:
For now, I just renamed the `TypedDict`s in https://github.com/deepset-ai/haystack-experimental/pull/319/commits/49dc95aa22a6d0ecf7103d0f3f34b13227a8c5b2.

If we end up reusing them in other modules (https://github.com/deepset-ai/haystack-experimental/pull/319#discussion_r2137461451), I will also move them.